File: src/main/java/com/cuupa/dms/storage/tag/db/MongoDBTagStorage.java
package com.cuupa.dms.storage.tag.db;

import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface MongoDBTagStorage extends MongoRepository<DBTag, Long> {

    List<DBTag> findTagsByOwner(String owner);

    //DBTag findTagById(long id);

    DBTag findTagByNameAndOwner(String name, String owner);
}

File: src/main/java/com/cuupa/dms/storage/tag/db/MongoDBTagStorage.kt
package com.cuupa.dms.storage.tag.db

import org.springframework.data.mongodb.repository.MongoRepository

interface MongoDBTagStorage : MongoRepository<DBTag?, Long?> {

    fun findTagsByOwner(owner: String): List<DBTag>
    //DBTag findTagById(long id);
    fun findTagByNameAndOwner(name: String, owner: String?): DBTag?
}